### ISSUE TITLE
No-unsafe-first not working as expected

### ISSUE THREAD

**Issue by hnipps** [cartant/eslint-plugin-rxjs#43]
The `no-unsafe-first` rule doesn't seem to work as expected.

I'm seeing ESLint errors when I use `take(1)` inside an Angular component but no error when I use it inside an epic. 

I'd expect to see the opposite: an error in the epic and no error in the component.

**Comment by timdeschryver** [comment]
Do you got some code you can share @hnipps ?
If I'm not mistaken the rule should look for the `take` on Observables with the [name `actions($)`](https://github.com/cartant/eslint-plugin-rxjs/blob/master/source/constants.ts#L6)

**Comment by hnipps** [comment]
```ts
export class MyComponent implements OnInit {

  ngOnInit(): void {
    this.transactionStateService.transactionType$.pipe(
      take(1), // ERROR: Unsafe first and take usage in effects and epics are forbidden. eslint(rxjs/no-unsafe-first)
      takeUntil(this.unsubscribe$)
    ).subscribe(transactionDisplay => {
      this.setDSBInfo(transactionDisplay);
    });
  }
}
```

Where `transactionType$` is a selector.

**Comment by timdeschryver** [comment]
fyi, I tried to reproduce it but wasn't able to - https://github.com/cartant/eslint-plugin-rxjs/compare/master...timdeschryver:issue-43?expand=1

**Comment by cartant** [comment]
@hnipps How have you configured the rule? As Tim has stated, it matches target observables by name using a regular expression: https://github.com/cartant/eslint-plugin-rxjs/blob/a58c81fd900c61ec864b8c3c916785b25a243130/source/rules/no-unsafe-first.ts#L68

And the default is: https://github.com/cartant/eslint-plugin-rxjs/blob/a58c81fd900c61ec864b8c3c916785b25a243130/source/constants.ts#L6

**Comment by cartant** [comment]
It's a bug - the regular expression needs a trailing `$` to avoid matching in the middle of an identifier.

**Comment by hnipps** [comment]
I'm just using the default config: `"rxjs/no-unsafe-first": "error";`

**Comment by timdeschryver** [comment]
🤦‍♂️- good catch.

**Comment by cartant** [comment]
Fixed in `0.0.2-beta.12`. Thanks for reporting it, @hnipps

**Comment by hnipps** [comment]
Thank you @cartant and @timdeschryver !
